File: Front-end/JS Fundamentals/arrays/10. Prime numbers/sieve-of-eratosthenes.js
function solve(args) {
    var i,
        j,
        n = +args,
        primes = new Array(n + 1),
        lengthPrimes = primes.length;

    for (i = 0; i < lengthPrimes; i += 1) {
        primes[i] = true; // array of boolean
    }

    // Find all prime numbers in the range [1...n]
    for (i = 2; i < Math.sqrt(lengthPrimes); i += 1) {
        // Skip those numbers that are not prime
        if (primes[i]) {
            for (j = i * i; j < lengthPrimes; j += i) {
                primes[j] = false;
            }
        }
    }

    for (i = lengthPrimes - 1; i >= 2; i--) {
        if (primes[i]) {
            console.log(i);
            break;
        }
    }

    // Print all prime numbers in the range [1...n]
    // process.stdout.write('2');
    // for (i = 3; i < lengthPrimes; i++) {
    //     // the numbers remaining not marked in the list are all the primes below n
    //     if (primes[i]) {
    //         process.stdout.write(', ' + i);
    //     }
    // }
    // console.log();

}

solve(13);
solve(126);
solve(26);

File: C#/C# 2/1. Arrays-HW/Prime-Numbers/SieveOfEratosthenes.cs
/*  Problem 15. Prime numbers
    Write a program that finds all prime numbers in the range [1 ... N]. 
    Use the Sieve of Eratosthenes algorithm. 
    The program should print the biggest prime number which is <= N.
    Input: On the first line you will receive the number N
    Output: Print the biggest prime number which is <= N
    Constraints: 2 <= N <= 10 000 000
 */

using System;

class SieveOfEratosthenes
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        bool[] primes = new bool[n + 1]; // False by default
        
        // Find all prime numbers in the range [1...n]
        for (int i = 2; i <= Math.Sqrt(n); i++) //n + 1 = primes.Length
        {
            // Skip those numbers that are not prime
            if (primes[i] == false)
            {
                for (int j = i * i; j <= n; j += i)
                {
                    primes[j] = true;
                }
            }
        }

        for (int i = n; i >= 2; i--)
        {
            if (!primes[i])
            {
                Console.WriteLine(i);
                break;
            }
        }

        //Print all prime numbers in the range[1...n]
        //for (int i = 2; i <= n; i++)
        //{
        //    // the numbers remaining not marked in the list are all the primes below n
        //    if (!primes[i])
        //    {
        //        Console.Write(i + " ");
        //    }
        //}
    }
}